File: dp_labs_and_solutions/dp_labs_and_solutions/dp_java_labs_and_solutions/observer/solution/Developer.java
// Developer.java

public class Developer extends Participant
{
	public void add(Meeting aMeeting)
	{
		super.add(aMeeting);

		System.out.println(
			  "Developer.add("
			+ aMeeting.toString()
			+ ")"
			);
	}

	public void remove(Meeting aMeeting)
	{
		super.remove(aMeeting);

		System.out.println(
			  "Developer.remove("
			+ aMeeting.toString()
			+ ")"
			);
	}

	public void update(Meeting aMeeting)
	{
		super.update(aMeeting);

		System.out.println(
			  "Developer.update("
			+ aMeeting.toString()
			+ ")"
			);
	}
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_matlab_labs_and_solutions/observer/solution/Developer.m
% Developer.m

classdef Developer < Participant
	methods
		function add(theDeveloper, aMeeting)
			theDeveloper.add@Participant(aMeeting);

			fprintf('Developer.add(%s)\n', aMeeting.toString);
		end

		function remove(theDeveloper, aMeeting)
			theDeveloper.remove@Participant(aMeeting);

			fprintf('Developer.remove(%s)\n', aMeeting.toString);
		end

		function update(theDeveloper, aMeeting)
			theDeveloper.update@Participant(aMeeting);

			fprintf('Developer.update(%s)\n', aMeeting.toString);
		end
	end
end
